components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java

```diff
@@ -3412,25 +3412,49 @@
         public static final String GET_MOST_RECENT_REVISION_UUID = "SELECT REVISION_UUID FROM AM_REVISION WHERE " +
                 "API_UUID = ? ORDER BY ID DESC LIMIT 1";
         public static final String GET_REVISION_APIID_BY_REVISION_UUID = "SELECT API_UUID, ID FROM AM_REVISION WHERE REVISION_UUID = ?";
-        public static final String GET_REVISIONS_BY_API_UUID = "SELECT * FROM AM_REVISION AR LEFT JOIN AM_DEPLOYMENT_REVISION_MAPPING ADRM"
-                + " ON AR.REVISION_UUID = ADRM.REVISION_UUID  WHERE AR.API_UUID = ? ";
+        static final String GET_API_REVISION_DEPLOYMENTS
+                = "(SELECT t3.NAME, t3.VHOST, t3.REVISION_UUID, MAX(t3.DEPLOYED_TIME) AS DEPLOYED_TIME, " +
+                "MAX(t3.DISPLAY_ON_DEVPORTAL) AS DISPLAY_ON_DEVPORTAL, MAX(t3.DEPLOY_TIME) AS DEPLOY_TIME " +
+                "FROM " +
+                "(SELECT NAME, VHOST, REVISION_UUID, DEPLOYED_TIME, NULL AS DISPLAY_ON_DEVPORTAL, NULL AS DEPLOY_TIME " +
+                "FROM " +
+                "AM_DEPLOYED_REVISION t1 " +
+                "UNION " +
+                "SELECT NAME, VHOST, REVISION_UUID, NULL AS DEPLOYED_TIME, DISPLAY_ON_DEVPORTAL, DEPLOYED_TIME as DEPLOY_TIME " +
+                "FROM AM_DEPLOYMENT_REVISION_MAPPING t2) AS t3 " +
+                "GROUP BY t3.NAME, t3.REVISION_UUID, t3.VHOST)";
```

Comment:
Shall we use first charactors of the table rather than using t1, t2.. ?
eg: AM_DEPLOYED_REVISION -> DR